### PR TITLE
Added x86 'sha256/sha512' detection.

### PR DESCRIPTION
**Pull request by whyb**
## Added x86 'sha256/sha512' detection.
x86 sha isa reference: [Intel_SHA_extensions Wikipedia](https://en.wikipedia.org/wiki/Intel_SHA_extensions)
圖片

So I rename x86 'sha' to 'sha1'.

## Test
godbolt test:
圖片

### CODE REVIEW

File: ruapu.h [hunk 1]
```diff
@@ -230,7 +230,9 @@ RUAPU_INSTCODE(bmi2, 0xc4, 0xe2, 0x7b, 0xf6, 0xc0) // mulx eax,eax,eax
 RUAPU_INSTCODE(gfni, 0x66, 0x0f, 0x38, 0xcf, 0xc0) // gf2p8mulb xmm0,xmm0
 RUAPU_INSTCODE(aesni, 0x66, 0x0f, 0x38, 0xdc, 0xc0) // aesenc xmm0,xmm0
 RUAPU_INSTCODE(vaes, 0xc4, 0xe2, 0x7d, 0xdc, 0xc0) // vaesenc ymm0,ymm0,ymm0
-RUAPU_INSTCODE(sha, 0x0f, 0x38, 0xc9, 0xc0) // sha1msg1 xmm0,xmm0
+RUAPU_INSTCODE(sha1, 0x0f, 0x38, 0xc9, 0xc0) // sha1msg1 xmm0,xmm0
+RUAPU_INSTCODE(sha256, 0x0f, 0x38, 0xcc, 0xc0) // sha256msg1 xmm0, xmm0
+RUAPU_INSTCODE(sha512, 0xc4, 0xe2, 0x7f, 0xcd, 0xd1) // vsha512msg2 ymm1, ymm2
```
Comment (nihui) [inline ruapu.h:235]: 尽量使用 xmm0 ymm0 zmm0

File: README.md [hunk 2]
```diff
@@ -784,7 +784,7 @@ ruapu determines whether the CPU supports certain instruction sets by trying to
 * [@dreamcmi](https://github.com/dreamcmi) &emsp;_Detect more risc-v ISA_
 * [@cocoa-xu](https://github.com/cocoa-xu) &emsp;_Add FreeBSD support, python support_
 * [@YuzukiTsuru](https://github.com/YuzukiTsuru) &emsp;_Add OpenRISC support_
-* [@whyb](https://github.com/whyb) &emsp;_Detect x86 AMX_
+* [@whyb](https://github.com/whyb) &emsp;_Detect more x86 AMX_, SHA_, AVX512_
```
Comment (nihui) [inline README.md:787]: 中间的下划线没有用吧

**Review by nihui — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (whyb) [reply README.md]: > 中间的下划线没有用吧

I have made the changes according to your request.

New:
* [@whyb](https://github.com/whyb) &emsp;_Detect more x86 AMX*, SHA*, AVX512*, SM*

File: ruapu.h (hunk 1 above)
Comment (whyb) [reply ruapu.h]: I have made the changes according to your request.
```c++
RUAPU_INSTCODE(sha1, 0x0f, 0x38, 0xc9, 0xc0) // sha1msg1 xmm0,xmm0
RUAPU_INSTCODE(sha256, 0x0f, 0x38, 0xcc, 0xc0) // sha256msg1 xmm0, xmm0
RUAPU_INSTCODE(sha512, 0xc4, 0xe2, 0x7f, 0xcd, 0xc0) // vsha512msg2 ymm0, ymm0
```
圖片

**Comment by nihui** [comment]
please fix the conflicts ...

**Comment by whyb** [comment]
> please fix the conflicts ...

Fixed conflicts.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -191,3 +191,12 @@ jobs:
...
           /bin/bash -c "apt update && apt install -y gcc libc6-dev && gcc main.c -o ruapu && ./ruapu"
+
+  macos-flyci:
+    runs-on: flyci-macos-14-xlarge-m2
+    steps:
+    - uses: actions/checkout@v4
+    - name: build-test
+      run: |
+        clang main.c -o ruapu
+        ./ruapu
```
Comment (nihui) [inline .github/workflows/ci.yml:202]: There is no flyci now

**Review by nihui — CHANGES_REQUESTED** [review]

**Comment by nihui** [comment]
Thanks for your contribution !

**Pull request merged**